capz/readme.md

```diff
@@ -4,44 +4,96 @@ These scripts and templates are used with [cluster-api-provider-azure](https://g
+# to the registry specified below.  Ensure you are logged in to the registry
+# prior to running the script!
+# TODO: Figure out how to use pre-built images here...
+export REGISTRY="<registry>"
 
 # optional for ability to use your own ssh key (otherwise it generates one)
 # NOTE: Azure does not support ed25519 encrypted SSH keys!
 export AZURE_SSH_PUBLIC_KEY_FILE="$HOME/.ssh/id_rsa.pub"
 ```
 
-## Other configuration
+#### Optional
 
 | ENV variable  | Description  |
 | ------------- | ------------ |
+| `API_SERVER_FEATURE_GATES` | Comma-separated list of feature-gates and their values to pass to the kube-apiserver |
+| `AZURE_LOCATION` | The azure region to deploy resources into |
+| `KUBERNETES_VERSION`  | Valid values are `latest` (default) and  `latest-1.xx` where x is valid kubernetes minor version such as `latest-1.24` |
+| `NODE_FEATURE_GATES` | Comma-seperated list of feature-gates and their values to pass to the kubelet |
+| `NODE_MACHINE_TYPE` | The [Azure vm size](https://learn.microsoft.com/en-us/azure/virtual-machines/sizes) to use for the nodes |
+| `RUN_SERIAL_TESTS` | If set to `true` then serial slow tests will be run with default ginkgo settings |
 | `SKIP_CREATE` | Don't create a cluster.  Must set `CLUSTER_NAME` and have current a workload cluster kubeconfig file with name `./"${CLUSTER_NAME}".kubeconfig` |
 | `SKIP_LOG_COLLECTION` | Don't collect logs from the cluster |
 | `SKIP_TEST`  | Only creates the cluster, will not run tests |
 | `SKIP_CLEANUP` | Don't delete the cluster / resource group after script executions |
-| `RUN_SERIAL_TESTS` | If set to `true` then serial slow tests will be run with default ginkgo settings |
-| `KUBERNETES_VERSION`  | Valid values are `latest` (default) and  `latest-1.x` where x is valid kubernetes minor version such as `latest-1.24` |
-| `AZURE_LOCATION` | The azure region to deploy resources into |
+| `SCHEDULER_FEATURE_GATES` | Comma-separated list of feature-gates and their values to pass to the kube-scheduler |
+| `WINDOWS_CONTAINERD_URL` | URL to a containerd release tarball to use for the Windows nodes |
 | `WINDOWS_KPNG` | If specified, will create a cluster using an out-of-tree kube-proxy implementation from [k-sigs/windows-service-proxy](https://github.com/kubernetes-sigs/windows-service-proxy) |
+| `WINDOWS_SERVER_VERSION` | Set to `windows-2019` (default) or `windows-2022` to test Windows Server 2019 or Windows SErver 2022 |
+| `WINDOWS_WORKER_MACHINE_COUNT` | Number of **Windows** worker nodes to provision in the cluster
+| `WORKER_MACHINE_COUNT` | Number of **Linux** worker nodes to provision in the cluster |
 
 ## GMSA support
 
 Set the environment variable `GMSA=true`.
 
 This requires additional set up in the Azure Subscriptions. See the readme in [gmsa folder](gmsa/readme.md).
 
+## Testing custom Kubernetes components
+
+`run-capz-e2e.sh` can privision clusters with custom Kubernetes components built from a local K8s repository.
```

Comment:
How do we destroy the provisioned cluster?

Reply:
if its the workload cluster, connect to management and run `kubectl delete cluster <capz-conf-name>`.  If you want to delete the management cluster delete the resource group that the AKS cluster is running in.

Reply:
@jsturtevant did you mean `clusterctl delete cluster`?

Reply:
Oh did not realize there is a CRD called `clusters.cluster.x-k8s.io`.